burn-tensor/src/tensor/api/base.rs

```diff
@@ -110,6 +110,50 @@ where
         Tensor::new(K::reshape::<D, D2>(self.primitive, new_dims.into()))
     }
 
+    /// Squeeze the tensor along the given dimension, removing the specified dimension
+    /// of size one, and effectively reducing the rank of the tensor by one.
+    ///
+    /// # Arguments
+    ///
+    /// - `dim`: The dimension to be squeezed.
+    ///
+    /// # Type Parameters
+    ///
+    ///  - 'D2': The resulting number of dimensions in the squeezed tensor.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor<B, D2, K>` instance with the specified dimenension removed.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    ///
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Tensor, Shape};
+    ///
+    /// fn example<B: Backend>() {
+    ///     let tensor = Tensor::<B, 3>::ones(Shape::new([2, 1, 4]));
+    ///
+    ///     // Given a 3D tensor with dimensions (2, 1, 4), squeeze the dimension 1
+    ///     let squeezed_tensor: Tensor::<B, 2> = tensor.squeeze::<2>(1);
+    ///
+    ///     // Resulting tensor will have dimensions (2, 4)
+    ///     println!("{:?}", squeezed_tensor.shape());
+    /// }
+    /// ```
+    pub fn squeeze<const D2: usize>(self, dim: usize) -> Tensor<B, D2, K> {
```

Comment:
Missing unit tests. Please see how they are added in this PR for `flatten` functionality. It was not straightforward for me, so I am listing the steps here.

https://github.com/burn-rs/burn/pull/260/files

1. add test under `burn-tensor/src/tests/ops`
2. add module under `burn-tensor/src/tests/ops/mod.rs`
3. modify burn-tensor/src/tests/mod.rs and add macro call

Reply:
@antimora Thanks, this was super helpful! As a side note it seems some ops are missing unit tests (like unsqueeze). Just though I'd bring it up.

Reply:
Thanks, @agelas. I have filed a task ticket to add the missing unit tests: https://github.com/burn-rs/burn/issues/384